### ISSUE TITLE
Cell formatted as a number with a hyperlink still renders as text

### ISSUE THREAD

**Issue by arknotts** [EPPlusSoftware/EPPlus#639]
I'd like to have a cell formatted as a number that is also a hyperlink to another cell within the Excel file. As far as I can tell, the only way to link to another cell within the Excel file is to new up a `ExcelHyperlink` object, and set that on the `Hyperlink` property of the cell. However, `ExcelHyperlink` requires that you pass the cell value as a string and it never gets formatted as a number in the Excel file output. Example code:

```
using OfficeOpenXml;

const string NUMBER_FORMAT = "$#,##0.00_);($#,##0.00)";

using (var excelPackage = new ExcelPackage("output.xlsx"))
{
	var worksheet = excelPackage.Workbook.Worksheets.Add("Test");

	var cellWithoutHyperlink = worksheet.Cells[1, 1];
	var cellWithHyperlink = worksheet.Cells[2, 1];
	var hyperlinkTargetCell = worksheet.Cells[3, 1];
	var numberValue = 100;

	cellWithoutHyperlink.Style.Numberformat.Format = NUMBER_FORMAT;
	cellWithoutHyperlink.Value = numberValue;
	cellWithHyperlink.Style.Numberformat.Format = NUMBER_FORMAT;
	cellWithHyperlink.Value = numberValue;

	cellWithHyperlink.Hyperlink = new ExcelHyperLink(hyperlinkTargetCell.FullAddress, numberValue.ToString());
	excelPackage.Save();
}
```

The resulting output:
![image](https://user-images.githubusercontent.com/705898/167203320-4ad74600-1c67-4409-a13c-789924d7bc1c.png)

**Comment by JanKallman** [comment]
Yes, EPPlus override the value in the cell when setting the hyperlink...
https://github.com/EPPlusSoftware/EPPlus/blob/develop/src/EPPlus/ExcelRangeBase.cs#L320
I'll see if I can add a new overload without the required display parameter to allow the value in the cell to be used instead. 
A workaround for now is to set the value and number format of the cell after adding the hyperlink.

**Comment by JanKallman** [comment]
I'll see if we can fix so it works in Google Sheets as well.
